bitbots_head_mover/src/move_head.cpp

```diff
@@ -0,0 +1,679 @@
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp/clock.hpp>
+#include <rclcpp/time.hpp>
+#include <rclcpp/logger.hpp>
+
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <sensor_msgs/msg/joint_state.hpp>
+#include <bitbots_msgs/msg/joint_command.hpp>
+#include <humanoid_league_msgs/msg/head_mode.hpp>
+
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/robot_state/conversions.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_broadcaster.h>
+#include <tf2/convert.h>
+#include <bio_ik/bio_ik.h>
+#include <bio_ik_msgs/msg/ik_response.hpp>
+
+#include "head_parameters.hpp"
+
+#include <rclcpp_action/rclcpp_action.hpp>
+#include <bitbots_msgs/action/look_at.hpp>
+
+
+using std::placeholders::_1;
+using namespace std::chrono_literals;
+
+namespace move_head {
+  using LookAtGoal = bitbots_msgs::action::LookAt;
+  using LookAtGoalHandle = rclcpp_action::ServerGoalHandle<LookAtGoal>;
+
+
+class HeadMover {
+  std::shared_ptr<rclcpp::Node> node_;
+
+  //declare subscriber and publisher
+  rclcpp::Subscription<humanoid_league_msgs::msg::HeadMode>::SharedPtr head_mode_subscriber_;
+  rclcpp::Subscription<sensor_msgs::msg::JointState>::SharedPtr joint_state_subscriber_;
+  rclcpp::Subscription<geometry_msgs::msg::PoseWithCovarianceStamped>::SharedPtr ball_subscriber_;
```

Comment:
The head mover is not interested in the ball anymore right? The body behavior sends look at goals with ball coordinates, but thats it.